### PR TITLE
[RBAC] az role definition create: print human readable message instead of exception when assignableScope is an empty array

### LINKED ISSUE #13610: az crashes when creating a role definition
## Describe the bug

**Command Name**
`az role definition create`

**Errors:**
```
list index out of range
Traceback (most recent call last):
Temp\pip-install-umfcm_n2\knack\knack\cli.py, ln 206, in invoke
azure\cli\core\commands\__init__.py, ln 603, in execute
azure\cli\core\commands\__init__.py, ln 661, in _run_jobs_serially
azure\cli\core\commands\__init__.py, ln 654, in _run_job
Local\Temp\pip-install-umfcm_n2\six\six.py, ln 693, in reraise
azure\cli\core\commands\__init__.py, ln 631, in _run_job
azure\cli\core\commands\__init__.py, ln 306, in __call__
azure-cli-core\azure\cli\core\__init__.py, ln 485, in default_command_handler
azure\cli\command_modules\role\custom.py, ln 51, in create_role_definition
azure\cli\command_modules\role\custom.py, ln 104, in _create_update_role_definition
azure\cli\command_modules\role\_multi_api_adaptor.py, ln 41, in create_role_definition
IndexError: list index out of range
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

```
$ type .\ResourceProviderRole.json
{
  "Name": "ACS Resource Provider",
  "IsCustom": true,
  "Description": "Act as the Azure Communication Services resource provider service",
  "Actions": [
    "Microsoft.Communication/*/read"
  ],
  "NotActions": [],
  "DataActions": [],
  "NotDataActions": [],
  "AssignableScopes": []
}

$ az role definition create --role-definition .\ResourceProviderRole.json
```

## Environment Summary
```
Windows-10-10.0.19612-SP0
Python 3.6.6
Shell: powershell.exe

azure-cli 2.0.76 *

```

### PR DESCRIPTION
**Pull request by arrownj**
**Description**  
This PR is to fix #13610 

Currently, if the role definition is below json:
>{
  "Name": "ACS Resource Provider",
  "IsCustom": true,
  "Description": "Act as the Azure Communication Services resource provider service",
  "Actions": [
    "Microsoft.Communication/*/read"
  ],
  "NotActions": [],
  "DataActions": [],
  "NotDataActions": [],
  "AssignableScopes": []
}
CLI will throw an exception and print the stack trace out directly.

The code actually has a check for `AssignableScopes`, however not handle the empty array case.

This fix is just add check for this case.

**Testing Guide**  
1. create a json file which has above content, call it `role.json` under current workspace.

2. run `az role definition create --role-deifnition ./role.json`

Expected output:
>please provide 'assignableScopes'

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RBAC

**Comment by yungezz** [comment]
does rest api accept empty array? if service accepts, seems should fix CLI code of consuming the json.

**Comment by arrownj** [comment]
> does rest api accept empty array? if service accepts, seems should fix CLI code of consuming the json.

The scope is used in the  REST api endpoint, so it is a required field here.

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def _create_update_role_definition(cmd, role_definition, for_update):
         if not role_name:
             raise CLIError("please provide role name")
 
-    if not for_update and 'assignableScopes' not in role_definition:
+    if not for_update and not('assignableScopes' in role_definition and role_definition['assignableScopes']):
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/custom.py:102]:
```suggestion
    if not for_update and not(role_definition.get('assignableScopes')):
```

Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/role/custom.py]: Good point. `assignableScopes` can actually be omitted, use `role_definition.get('assignableScopes', None)` instead.

**Pull request merged**